File: MAP/Lab10/ConsoleApp1/ConsoleApp1/Model/statements/PrintStmt.cs
using ConsoleApp1.Model.state;
using ConsoleApp1.Model.statements;
using System;
using System.Collections.Generic;
using System.Text;
using ToyInterpreter.Model.expressions;

namespace ToyInterpreter.Model.statements
{
    class PrintStmt: IStmt
    {
        private IExp exp;

        public PrintStmt(IExp exp)
        {
            this.exp = exp;
        }

        public PrgState execute(PrgState state)
        {
            List<Int64> printState = state.getOut();
            printState.Add( exp.eval(state.getSymTable()) );

           return state;
        }

        public override String ToString()
        {
            return "print(" + exp.ToString() + ")";
        }

}
}

File: MAP/PracticalExamPractice/03.ProcedureToyInterpreter/src/model/statements/PrintStmt.java
package model.statements;

import model.adts.MyIList;
import model.expressions.IExp;
import model.state.PrgState;

public class PrintStmt implements IStmt{
    private IExp exp;

    public PrintStmt(IExp exp) {
        this.exp = exp;
    }

    public PrgState execute(PrgState state) throws Exception {
        MyIList<Integer> printState = state.getOut();
        printState.push_back(exp.eval(state.getTopSymTable(), state.getHeap()) );

        return null;
    }

    @Override
    public String toString() {
        return "print(" +exp.toString()+")";
    }

}